com.github.sdbg.debug.core/src/com/github/sdbg/debug/core/internal/util/DebuggerUtils.java

```diff
@@ -80,39 +80,7 @@ public static String demangleVmName(String name) {
       return null;
     }
 
-    int atIndex = name.indexOf('@');
-
-    while (atIndex != -1) {
-      // check for _foo@76876.bar (or _Process@14117cc4._reportError@14117cc4)
-      int endIndex = name.indexOf('.', atIndex);
-
-      if (endIndex == -1) {
-        name = name.substring(0, atIndex);
-      } else {
-        name = name.substring(0, atIndex) + name.substring(endIndex);
-      }
-
-      atIndex = name.indexOf('@');
-    }
-
-    // Also remove the trailing '.' for default constructors.
-    if (name.endsWith(".")) {
-      name = name.substring(0, name.length() - 1);
-    }
-
-    // remove "set:" and "get:"
-    // Cat.get:color() ==> Cat.color()
-    if (name.indexOf(".set:") != -1) {
-      int index = name.indexOf(".set:");
-      name = name.substring(0, index + 1) + name.substring(index + 5);
-    }
-
-    if (name.indexOf(".get:") != -1) {
-      int index = name.indexOf(".get:");
-      name = name.substring(0, index + 1) + name.substring(index + 5);
-    }
-
-    return name;
+    return name.replaceAll("_[0-9]+_g\\$", "");
```

Comment:
ok I'll fix and force it onto this same commit.  Might take a few hours.

On Sun, Jan 18, 2015 at 3:54 PM, ivmarkov <notifications@github.com> wrote:

> In
> com.github.sdbg.debug.core/src/com/github/sdbg/debug/core/internal/util/DebuggerUtils.java
> <https://github.com/sdbg/sdbg/pull/106#discussion-diff-23133414>:
>
> > -    }
> > -
> > -    // remove "set:" and "get:"
> > -    // Cat.get:color() ==> Cat.color()
> > -    if (name.indexOf(".set:") != -1) {
> > -      int index = name.indexOf(".set:");
> > -      name = name.substring(0, index + 1) + name.substring(index + 5);
> > -    }
> > -
> > -    if (name.indexOf(".get:") != -1) {
> > -      int index = name.indexOf(".get:");
> > -      name = name.substring(0, index + 1) + name.substring(index + 5);
> > -    }
> > -
> > -    return name;
> > +    return name.replaceAll("_[0-9]+_g\\$", "");
>
> I think this pattern will work throughout the whole string, while we want
> that it only works for the suffix of the string, which is not enforced. To
> enforce it, it should be "*[0-9]+_g\$$" or "*[0-9]+_g\$\z" or something
> similar.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/sdbg/sdbg/pull/106/files#r23133414>.
>